src/dns/coredns.rs

```diff
@@ -138,6 +138,77 @@ impl CoreDns {
                                 self.kill_switch.lock().unwrap()
                             );
 
+                            // if record type is PTR try resolving early and return if record found
+                            if record_type == RecordType::PTR {
```

Comment:
This is very deeply nested code, seems like some of this could be broken out into helper fuctions that could also be unit tested e.g.